src/v2/view-builder/views/AdminConsent.js

```diff
@@ -0,0 +1,82 @@
+import { loc, View, Model } from 'okta';
+import BaseView from '../internals/BaseView';
+import BaseForm from '../internals/BaseForm';
+import ScopeList from '../../../views/admin-consent/ScopeList';
+import consentLogoHeaderTemplate from '../../../views/shared/templates/consentLogoHeaderTemplate';
+
+
+const Header = View.extend({
+  className: 'consent-title detail-row',
+  template: consentLogoHeaderTemplate,
+  getTemplateData: function () {
+    const {label: appName, clientUri = {} } =  this.options.appState.get('app');
+    const customLogo = this.settings.get('logo');
+    const defaultLogo = '/img/logos/default.png';
+    const clientURI = clientUri.href;
+    const { issuer: issuerObj = {} } = this.options.appState.get('authentication');
+    const issuer = issuerObj.uri;
+
+    return {
+      appName, // escape?
+      customLogo,
+      defaultLogo,
+      clientURI,
+      issuer,
+    };
+  }
+});
+
+const Body = BaseForm.extend(
+  {
+    noButtonBar: false,
+    noCancelButton: false,
+    save: () => loc('consent.required.consentButton', 'login'),
+    cancel: () => loc('consent.required.cancelButton', 'login'),
+
+    /**
+     * Format scopes for ScopeList and render ScopeList
+     */
+    initialize () {
+      BaseForm.prototype.initialize.apply(this, arguments);
+      this.uiSchema = this.getUISchema();
+
+      const scopes = this.uiSchema.map(({label, desc}) => {
+        return {name: label, displayName: label, description: desc};
+      });
+      const model = new Model({ scopes });
+      this.add(ScopeList, {options: { model }});
+    },
+    /**
+     * Format scopes to match the schema required by the server
+     * POST /idp/idx/consent
+        {
+          "scopes": [ {{value}} ]
+        }
+     */
+    saveForm () {
```

Comment:
- Did you double check the request body to see it didn't send extra data?
- Looks like we shall override `BaseView.createModelClass`
- In additions, feel like we shall update `ion-object-handler#createUiSchemaForObject` to create a customized view for `scopes` field. Similar to how we handle `select-authenticator-authenticate`. @pradeepdewda-okta @nikhilvenkatraman-okta any thoughts?

Reply:
@haishengwu-okta 
yeah i would discourage to override saveForm. `scopes` is a new attribute for v2, so yes that feels right place to add this.

Reply:
But i also think that could be overkill to do this? It's a specific use case and not reusable..

Reply:
Note: `scopes` has always been part of `authParams`, and that it now exists OUTSIDE of authParams is likely behavior we don't want to keep.  We have a bug about this issue.

Reply:
`scopes` is just field name in a remediation form (which will be show up in request body).
why it's being conflict/confused with `authParams` (OIDC `scope`) ?

Reply:
^ @swiftone, the scopes here are for a form that will be submitted to /idx/consent (if the user accepts these scopes) and then those scopes will be incorporated in the state token. I believe the authParams would be referring to the accepted scopes in the token?

Reply:
I am sending the correct data on saveForm (you can see PR giphy for what that is), but another way is to use an event listener instead like this https://github.com/okta/okta-signin-widget/pull/1637/commits/f83396c5cf57506e8cdff79e717997c71a7d8e9f#diff-8a1abd3d64283eaee376fa7479ddf4070fbfaa7c93b59fe3164d2be7521a6394R34

Reply:
Nice see, if you can trigger a event.. Also as mentioned by @haishengwu-okta super method should always be called first and then override statements .. ```BaseForm.prototype.saveForm.apply(this, arguments);```